packages/@atjson/renderer-commonmark/src/index.ts

```diff
@@ -717,4 +737,124 @@ export default class CommonmarkRenderer extends Renderer {
     let rawText = yield;
     return rawText.join("");
   }
+
+  *DataSet(): Generator<void, string, string[]> {
+    return "";
+  }
+
+  *Table(
+    table: Block<Table>,
+    context: Context
+  ): Generator<void, string, string[]> {
+    if (this.state.inlineOnly) {
+      return "";
+    }
+
+    const previousState = this.state;
+    this.state.inlineOnly = true;
+
+    const dataSet = context.document.blocks.find(
+      (block) => block.id === table.attributes.dataSet
+    ) as Block<DataSet> | undefined;
+
+    if (!dataSet) {
+      /**
+       * invalid dataset ref
+       */
+
+      throw new Error(
+        `table ${table.id} references nonexistent dataset ${table.attributes.dataSet}`
+      );
+    }
+
+    const columns: Record<
+      string,
+      {
+        header: string;
+        rows: string[];
+        width: number;
+        textAlign?: "left" | "center" | "right";
+      }
+    > = {};
+
+    for (let { name, slice: sliceId, textAlign } of table.attributes.columns) {
+      let headerText = "";
+      if (table.attributes.showColumnHeaders) {
+        if (sliceId) {
+          const slice = this.getSlice(sliceId);
+          if (!slice) {
+            throw new Error(`column heading slice not found ${sliceId}`);
+          }
+          headerText = this.render(slice);
+        } else {
+          headerText = name;
+        }
+      }
+      columns[name] = {
+        header: headerText,
+        rows: [],
+        width: Math.max(headerText.length, 1),
+        textAlign,
+      };
+    }
+
+    for (let row of dataSet.attributes.records) {
+      for (let { name } of table.attributes.columns) {
+        let cellText = "";
+        let sliceId = row[name]?.slice;
+
+        if (sliceId) {
+          let cellSlice = this.getSlice(sliceId);
+          if (!cellSlice) {
+            throw new Error(
+              `Table ${table.id} ${table.range} with DataSet ${
+                dataSet.attributes.name || dataSet.id
+              }: document slice not found for column ${name} in row ${JSON.stringify(
+                row,
+                null,
+                2
+              )}`
+            );
+          }
+          cellText = this.render(cellSlice);
+        }
+
+        columns[name].rows.push(cellText);
+        columns[name].width = Math.max(columns[name].width, cellText.length);
+      }
+    }
+
+    let headerRow = "|";
+    let separatorRow = "|";
+
+    for (let { name, textAlign } of table.attributes.columns) {
+      let headerText = columns[name].header;
+      let columnWidth = columns[name].width;
+      headerRow +=
+        " " + headerText + " ".repeat(columnWidth - headerText.length) + " |";
+
+      let leftDecoration =
+        textAlign === "left" || textAlign === "center" ? ":" : " ";
```

Comment:
I would personally use another dash for when it's not `:` but this works! The symmetry is kinda nice.